### ISSUE TITLE
[Footer]: QA: Selected option is visually displayed but not identified by the screen reader

### ISSUE THREAD

**Issue by Gopi916** [carbon-design-system/carbon-for-ibm-dotcom#8854]
### Description

Page Element:
Select geographic area button that triggers the modal

Issue:
Currently selected option is visually displayed but not identified by the screen reader

Expected:
Along with the button label, screen reader is expected to announce the currently selected option

User Impact:
If the screen reader fails to identify the option selected, it impacts user interactivity.

### Component(s) impacted

Footer (Default)

### Browser

Firefox

### Carbon for IBM.com version

v1.34.0-rc.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://www.ibm.com/standards/carbon/web-components/?path=/story/

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
